test/highlvl-async-usage.test.js

```diff
@@ -172,6 +172,25 @@ describe('basic usage of high-level async methods', function(){
 
   });//</it should have aggregated some config as expected>
 
+  it('should maintain object references when merging modules together', function(done) {
+
+    loader.aggregate({
+      dirname: path.resolve(__dirname, './fixtures/test-refs/config'),
+      filter: /(.+)\.js$/
+    }, function (err, config){
+      if (err) { return done(err); }
+
+      assert(config.databases.someDb);
```

Comment:
Yeah not sure about Mocha, but I checked the output when the tests fail and they look reasonable as-is, so I don't think the try/catch is necessary in this case.  I did add some comments in the test to try and remind ourselves what it's actually checking.